Library/src/Multiplayer/SpaceEntity.cpp

```diff
@@ -179,11 +180,26 @@ const SpaceTransform& SpaceEntity::GetTransform() const
 	return Transform;
 }
 
+SpaceTransform SpaceEntity::GetGlobalTransform() const
+{
+	if (Parent != nullptr)
+		return Parent->GetTransform() * Transform;
+	return Transform;
+}
+
 const csp::common::Vector3& SpaceEntity::GetPosition() const
 {
 	return Transform.Position;
 }
 
+csp::common::Vector3 SpaceEntity::GetGlobalPosition() const
+{
+	if (Parent != nullptr)
+		return Parent->GetTransform().Position + Transform.Position;
```

Comment:
Does the global position not need to take into account the parents rotation?

Reply:
Doesn't scale need to eb taken into account in some circumstances, too?

Reply:
Another issue with the above is we should be calling the parent's GetGlobalTransform here, as otherwise you're just returning the transform relative to the parent (which isn't necessarily the root of the hierarchy).